app/models/document.rb

```diff
@@ -208,6 +214,12 @@ def save
   def publish!
     handle_remote_error do
       update_type = self.update_type || 'major'
+
+      if not_published?
+        self.change_note = "First published."
+        self.save
+      end
```

Comment:
I think I'd independently arrived at the same conclusion, this approach seems most accurate and as you say, once we have the 'state_history' information we can make this cleaner.